metricflow/model/semantics/data_source_semantics.py

```diff
@@ -26,7 +26,7 @@
 logger = logging.getLogger(__name__)
 
 
-class DataSourceSemantics:
+class DataSourceSemantics(DataSourceSemanticsAccessor):
```

Comment:
It's best not to do this, because this makes DataSourceSemantics inherit all properties from the protocol, which means anything that isn't tagged as an `@abstractmethod` in the protocol is automatically "correct" as far as the typechecker is concerned.

Today this makes no practical difference with regard to typechecking behavior, but if anybody adds a bare property without a default value to the protocol the typechecker will pass and anything accessing that property may fail in the runtime.

Reply:
Changed this to be an ABC. Without it having something inherited, it was hard to tell in the IDE which functions needed to be updated in `DataSourceSemantics` when  `DataSourceSemanticsAccessor` changed.